src-built-in/components/advancedAppCatalog/src/stores/storeActions.js

```diff
@@ -235,8 +235,28 @@ async function addApp(id, cb = Function.prototype) {
 		delete appConfig.window.url;
 	}
 
-	if (typeof app.manifest !== "object") {
-		appConfig.manifest = { ...appConfig };
+	let manifest;
+	if (app.manifestType === "Finsemble") {
+		try {
+			//Attempt to parse a string as JSON
+			manifest = JSON.parse(app.manifest);
+		} catch(e) {
+			try {
+				//If parsing fails, assume the manifest is a valid url that will return JSON
+				const urlRes = await fetch(app.manifest, { method: "GET" });
+
+				//Attempt to parse the url response
+				manifest = await urlRes.json();
+			} catch(e) {
+				FSBL.Clients.Logger.system.warn(`${app.title || app.name} is missing a valid manifest or URI that delivers a valid JSON manifest. Unable to add app`);
```

Comment:
Add period at the end of "Unable to add app". Without a defined language structure to follow let's at least try to be consistent. You don't have punctuation at the end here where on the warning in 258 you do.